draft-add-requirements.md

```diff
@@ -32,89 +56,180 @@ informative:
+
+The client may wish to select an equivalent associated resolver, or select one of the alternatives.
+
+Designs for resolver upgrade mechanisms can either add new parameters to existing provisioning mechanisms
+(for example, adding necessary information to use DoT or DoH to options in DHCP, RAs, or IKEv2) or else provide a way
+to communicate with a provisioned unencrypted DNS resolver and discover the associated encrypted DNS resolvers.
 
 | Requirement | Description |
-| R1.1 | There must be a mechanism for a client to learn the set of encrypted resolvers that are associated with a resolver that is known only by its IP address. |
-| R1.2 | Discovery must be possible even when the IP address is only valid locally. |
+| R1.1 | There must be a mechanism for a client to learn the set of encrypted resolvers that are associated with an unencrypted resolver. |
+| R1.2 | Discovery must be possible even when the IP address of the encrypted resolver is only valid locally. |
 | R1.3 | More to be added |
 
-## Discover resolvers associated with one declared by an attached network
+## Network-provisioned resolvers {#local-network}
 
-During network attach, networks frequently announce a resolver that they recommend clients should use.
-This could be through DHCP, RA, PPP, PCO, or another mechanism. Historically this is usually a Do53 resolver,
-declared simply by its IP address. However it could also be a richer set of information.
+DNS servers are often provisioned by a network as part of DHCP options {{?RFC2132}},
+IPv6 Router Advertisement (RA) options {{?RFC8106}}, Point-to-Point Protocol (PPP) {{?RFC1877}},
+3GPP Protocol Configuration Options, or another mechanism. Historically this is usually one or
+more DNS resolver IP addresses, to be used for traditional unencrypted DNS. However it could also
+be a richer set of information.
+
+Using an encrypted and authenticated resolver that is associated to the one provisioned by the network
+can provide several benefits that are not possible if only unencrypted DNS is used:
+
+- Prevent other devices on the network from observing client DNS messages
+- Verify that answers come from the selected DNS resolver
+- Authenticate that the DNS resolver is the one provisioned by the network
+
+Frequently, network-provisioned resolvers are forwarders running on a local router. The discovered
+encrypted resolvers in these cases may either be local fowarders themselves, or an associated
+resolver that is in the network (thus bypassing the router's DNS forwarder).
 
 | Requirement | Description |
 | R2.1 | Example requirement |
 
 ### Unencrypted forwarder
 
-If the resolver announced by the network is a classic unencrypted forwarder, it is frequently the case that such forwarders are difficult to upgrade
```

Comment:
It is difficult to upgrade legacy routers but modern home routers can be easily upgraded (they support LXC containers, RDK software https://rdkcentral.com/ to easily update config and software on routers).

Reply:
Some modern home routers, yes. But it remains the case that there is a large installed based of difficult-to-upgrade routers, and this is currently the majority of routers in the home in the UK. So I think the text seems still valid.

Reply:
It depends on the ISP. I know Comcast-managed routers run RDK software (https://rdkcentral.com/). They can update configs & software on devices with fine-grained control (controlled release, A/B test, roll fwd/back, etc.) and with whatever frequency is necessary.

Reply:
So my intention here is that "frequently the case" means frequently in a *global* sense, not frequently in the context of a single ISP.

Reply:
I quoted Comcast as an example only. RDK https://rdkcentral.com/ is not specific to Comcast and used by several others (for example see https://www.alticelabs.com/en/531-altice-labs-innovates-by-launching-its-new-fiber-gateway-with-wi-fi-6-and-rdk-b.html, https://rdkcentral.com/tata-elxsis-vision-for-next-gen-rdk-cpes/).

Reply:
Tiru do you have evidence that such deployments constitute the majority of home routers on the planet?

Reply:
My point is growing number of homes are switching to modern routers to get home network security to protect the network from internal and external attacks (especially IoT devices). McAfee Secure Home Platform can run in LXC and is already deployed by several ISPs. Several other companies also offer home network security. As you know a network security service cannot be offered without the capability to upgrade to handle new threats and to add new security capabilities.

The requirement should discuss both legacy and modern routers (we should not ignore them).

Reply:
Agree we should cover all classes of home routers. I was thinking that such modern routers were covered by the "Encrypted Forwarder" section, since the obvious way forward is to do the upgrade to support encryption. If you think more text is needed to cover such routers, e.g. to handle when they are upgradeable but do not support encrypted operation, then please could you submit a PR with the relevant text?